odxtools/diaglayer.py

```diff
@@ -457,50 +459,29 @@ def get_communication_parameter(self, cp_id: str) \
 
     def get_receive_id(self) -> Optional[int]:
         """CAN ID to which the ECU listens for diagnostic messages"""
-        com_param = self.get_communication_parameter("ISO_15765_2.CP_UniqueRespIdTable")
-
+        com_param = self.get_communication_parameter("CP_UniqueRespIdTable")
         if com_param is None:
             return None
-
-        if self._enable_candela_workarounds:
-            # assume the parameter order used by CANdela studio.
-            # note that the parameter ordering actually used
-            # differs from the one of the COMPARAM fragment
-            # delivered by CANdela generated PDX files and that
-            # both are different from the one of the COMPARAM
-            # fragment included in the MCD2-D standard
-            try:
-                return int(com_param.value[2])
-            except ValueError:
-                return None
-        else:
-            # assume the parameter order specified by the COMPARAM
-            # fragment of the ASAM MCD2-D standard.
-            return int(com_param.value[1])
```

Comment:
@andlaus you may want to double check this, the new logic is name based so `enable_candela_workarounds` should not be needed anymore, but I don't actually have the names that the code was supposed to point to.

Reply:
this PR swaps `ecu.get_send_id()` and `ecu.get_receive_id()`. It might be an issue of CANdela generated files? (does it work correctly with your files?)